platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/style/sources/GeoJsonSource.java

```diff
@@ -197,6 +197,21 @@ public void setGeoJson(Feature feature) {
     nativeSetFeature(feature);
   }
 
+  /**
+   * Updates the GeoJson while performing most resource-intensive work on a background thread and notifies the listener
+   * by {@link OnGeoJsonSourceLoadedListener#onGeoJsonSourceLoaded()} when the job is in the same state that
+   * a synchronous update would return at.
+   * <p>
+   * There can only be one asynchronous update running. Any overlapping ones will be rejected.
```

Comment:
On gl-js we moved to a "coalescing" model for high frequency asynchronous `setData` calls (https://github.com/mapbox/mapbox-gl-js/pull/5902). It might be a nice strategy to use here to make this interface a little more ergonomic -- instead of the caller having to worry about whether there's one in progress, they can rely on these guarantees:

- Any currently running request will run to completion
- The last request submitted will run to completion
- Requests in the middle may get discarded without any action required on the part of the caller

Reply:
Thats a great suggestion @ChrisLoer ! @LukasPaczos I think the coalescing model would be useful for https://github.com/mapbox/mapbox-gl-native/pull/12509 as well.

Reply:
Interesting idea, let me try to find a way of doing that.